Source/WebCore/platform/graphics/cocoa/CMUtilities.mm

```diff
@@ -277,6 +282,179 @@ static CFStringRef convertToCMYCbCRMatrix(PlatformVideoMatrixCoefficients coeffi
+            // Column 4: Frame Sizes
+            // Duration in number of frames (Opus is always 48000Hz)
+            constexpr size_t frameSizeArray[] = {
+                480,
+                960,
+                1920,
+                2880,
+                480,
+                960,
+                1920,
+                2880,
+                480,
+                960,
+                1920,
+                2880,
+                480,
+                960,
+                480,
+                960,
+                120,
+                240,
+                480,
+                960,
+                120,
+                240,
+                480,
+                960,
+                120,
+                240,
+                480,
+                960,
+                120,
+                240,
+                480,
+                960,
+            };
+            if (config < std::size(frameSizeArray))
+                return frameSizeArray[config];
+
+            ASSERT_NOT_REACHED();
+            return 960; // The most common Opus frame duration.
+        }(config);
+        break;
+        }
+#endif
+#if ENABLE(VORBIS)
+    case kAudioFormatVorbis:
+        {
+        // The following calculation corresponds to the duration of the "finished audiodata"
+        // produced by the decoder from the current packet in its position within
+        // the stream, as documented by Xiph in the Vorbis I specification.
+        // It also corresponds to the delta in granule position of the packet within
+        // the same sequence of packets in an Ogg file, with the possible exception of
+        // the ultimate packet, which may be assigned a smaller delta for the purpose
+        // of trimming.
+        //
+        // Further, it corresponds to the production of AudioStreamPacketDescriptions
+        // by the Vorbis encoder that's part of the now discontinued XiphQT components
+        // for QuickTime. See https://www.xiph.org/quicktime/.
+        //
+        // We acknowledge that it differs from a similar calculation performed by FFmpeg,
+        // which forms the basis of sample tables that it writes for Vorbis in .mp4 and .mov.
+        // However, FFmpeg is known to write unreliable sample tables. See rdar://77516749.
+#define kVorbisPacketTypeFlag 0b00000001
+
+        auto leadingByte = *packet.data();
+        if (leadingByte & kVorbisPacketTypeFlag)
+            return 0; // Not an audio packet.
+
+        uint32_t modeIndex = (leadingByte >> 1) & m_vorbisModeMask;
+        if (modeIndex >= m_vorbisModeInfo->mModeCount)
+            return 0; // Invalid mode.
+
+        uint32_t blockSize = 0;
+        if (!(m_vorbisModeInfo->mModeFlags & (1 << modeIndex)))
+            blockSize = m_vorbisModeInfo->mShortBlockSize;
+        else
+            blockSize = m_vorbisModeInfo->mLongBlockSize;
+        // The first vorbis packet decoded doesn't output audible content, and should be 0. But CoreMedia will choke on this.
```

Comment:
0 isn't strictly correct even when the underlying media stack permits a 0 packet duration. What's really correct is: undetermined, because the true determination of the duration of any Vorbis packet requires one prior packet.